core-test/src/test/scala/org/bitcoins/core/api/CallbackTest.scala

```diff
@@ -17,13 +17,19 @@ class CallbackTest extends BitcoinSAsyncTest {
 
     val f1: Callback[Unit] = _ => {
       Thread.sleep(testTimeout.toMillis)
```

Comment:
You might want to use use `system.scheduler` to schedule the promise to be completed at `testTimeout` rather than using `Thread.sleep`. I'm not sure if this is related or not, but after merging #1578 maybe there can be some thread starvation going on with `Thread.sleep()`?